src/client.cpp

```diff
@@ -908,6 +905,56 @@ void CClient::Stop()
     SignalLevelMeter.Reset();
 }
 
+/// @method
+/// @brief Connects to strServerAddress
+/// @emit Connecting (strServerName) if the client wasn't running and SetServerAddr returned true.
+/// @param strServerAddress - the server address to connect to
+/// @param strServerName - the String argument to be passed to Connecting()
+/// @result true if client wasn't running and SetServerAddr returned true, false otherwise
```

Comment:
I don't like this.  `Connect` says "Go off and connect".  It should either emit "Connected" or "ConnectionFailed" signals, not _return_ a value.  The point is that ClientDlg and Client can be on separate threads and you need to use the slots/signals to communicate.

Reply:
Agree. Potentially a larger rewrite.